src/main/java/net/gensokyoreimagined/motoori/KosuzuUnderstandsEverything.java

```diff
@@ -112,23 +113,28 @@ private void onServerChatSend(PacketEvent event) {
 
         var json = message.getJson();
         var component = JSONComponentSerializer.json().deserialize(json); // Adventure API from raw JSON
-        var text = parser.getTextMessage(component, player);
+        var text = PlainTextComponentSerializer.plainText().serialize(component);
+        var actualMessage = parser.getTextMessage(text, player);
 
-        if (text != null) {
-            var uuid = database.addMessage(json, text);
+        TextComponent newComponent;
+        if (actualMessage != null) {
+            actualMessage = parser.removeUnwantedSyntax(message);
+            var uuid = database.addMessage(json, actualMessage);
 
-            var newComponent = component.hoverEvent(
+            newComponent = component.hoverEvent(
                     Component
                         .text(database.getTranslation("translate.hover", database.getUserDefaultLanguage(player.getUniqueId())))
                         .color(NamedTextColor.GRAY)
                 )
                 .clickEvent(
                     ClickEvent.runCommand("/kosuzu translate " + uuid.toString())
                 );
-
-            var newJson = JSONComponentSerializer.json().serialize(newComponent);
-            packet.getChatComponents().write(0, WrappedChatComponent.fromJson(newJson));
+        } else {
+            newComponent = component.content(parser.removeUnwantedSyntax(text));
         }
+
+        var newJson = JSONComponentSerializer.json().serialize(newComponent);
+        packet.getChatComponents().write(0, WrappedChatComponent.fromJson(newJson));
```

Comment:
Do you mean to create a new component? Use the `Component` interface's static method `text`.